js/src/views/Application/TabBar/tabBar.js

```diff
@@ -140,12 +190,115 @@ export default class TabBar extends Component {
         value={ isTest ? 'TEST' : netChain } />
       );
 
-    return this.renderLabel('status', bubble);
+    return this.renderLabel(label, bubble);
   }
 
   onActivate = (tab) => {
     const { router } = this.context;
+    const activeRoute = tab.props['data-route'];
```

Comment:
Not sure I like those options either. I remember that I spent time researching this and this seemed to be the "least hacky way" in my view :( Thinking about it again, I will try your option #1 though, it may be slightly less problematic. (Still not great)